### PR TITLE
Make signature a tuple

### PR DESCRIPTION
**Pull request by hweawer**

### CODE REVIEW

File: scripts/lib/spammer.ts [hunk 1]
```diff
@@ -4,11 +4,11 @@ import { randomInt, sleep } from "./utils";
 import { Address, DataBusClient } from "../../client";
 
 const abi = [
-  "event MessageDepositV1(address indexed guardianAddress, (bytes32 depositRoot, uint256 nonce, uint256 blockNumber, bytes32 blockHash, bytes signature, uint256 stakingModuleId, (bytes32 version) app) data)",
-  "event MessagePauseV2(address indexed guardianAddress, (bytes32 depositRoot, uint256 nonce, uint256 blockNumber, bytes32 blockHash, bytes signature, uint256 stakingModuleId, (bytes32 version) app) data)",
-  "event MessagePauseV3(address indexed guardianAddress, (uint256 blockNumber, bytes signature, (bytes32 version) app) data)",
+  "event MessageDepositV1(address indexed guardianAddress, (bytes32 depositRoot, uint256 nonce, uint256 blockNumber, bytes32 blockHash, (bytes32 r, bytes32 vs) signature, uint256 stakingModuleId, (bytes32 version) app) data)",
```
Comment (F4ever) [inline scripts/lib/spammer.ts:7]: Optional:
Maybe use order as in deposit buffered ether method

```
        address guardianAddress,
        uint256 blockNumber,
        bytes32 blockHash,
        bytes32 depositRoot,
        uint256 stakingModuleId,
        uint256 nonce,
        Signature calldata signature,
        (bytes32 version) app,
```

File: scripts/lib/spammer.ts [hunk 2]
```diff
@@ -4,11 +4,11 @@ import { randomInt, sleep } from "./utils";
...
+  "event MessagePauseV2(address indexed guardianAddress, (bytes32 depositRoot, uint256 nonce, uint256 blockNumber, bytes32 blockHash, (bytes32 r, bytes32 vs) signature, uint256 stakingModuleId, (bytes32 version) app) data)",
```
Comment (F4ever) [inline scripts/lib/spammer.ts:8]: Remove `depositRoot`, `nonce` and `blockHash`

File: scripts/lib/spammer.ts [hunk 3]
```diff
@@ -4,11 +4,11 @@ import { randomInt, sleep } from "./utils";
...
+  "event MessagePauseV3(address indexed guardianAddress, (uint256 blockNumber, (bytes32 r, bytes32 vs) signature, (bytes32 version) app) data)",
   "event MessagePingV1(address indexed guardianAddress, (uint256 blockNumber, (bytes32 version) app) data)",
-  "event MessageUnvetV1(address indexed guardianAddress, (uint256 nonce, uint256 blockNumber, bytes32 blockHash, uint256 stakingModuleId, bytes signature, bytes32 operatorIds, bytes32 vettedKeysByOperator, (bytes32 version) app) data)",
+  "event MessageUnvetV1(address indexed guardianAddress, (uint256 nonce, uint256 blockNumber, bytes32 blockHash, uint256 stakingModuleId, (bytes32 r, bytes32 vs) signature, bytes32 operatorIds, bytes32 vettedKeysByOperator, (bytes32 version) app) data)",
```
Comment (F4ever) [inline scripts/lib/spammer.ts:11]: Optional order (like in unvet method):
```
blockNumber,
blockHash,
stakingModuleId,
nonce,
operatorIds,
vettedKeysByOperator,
siganture,
metadata
```

Comment (F4ever) [reply scripts/lib/spammer.ts:11]: Operators and key indexes is an unlimited bytes list
```
bytes32 operatorIds -> bytes operatorIds
bytes32 vettedKeysByOperator -> bytes vettedKeysByOperator
```

File: scripts/lib/spammer.ts (hunk 2 above)
Comment (eddort) [reply scripts/lib/spammer.ts:8]: why? how do we know that this message has already been processed without nonce?

File: scripts/lib/spammer.ts (hunk 1 above)
Comment (eddort) [reply scripts/lib/spammer.ts:7]: current order as in Council Daemon, do you think you should redo it?

File: scripts/lib/spammer.ts (hunk 2 above)
Comment (eddort) [reply scripts/lib/spammer.ts:8]: My suggestion is to keep `blockHash` for debugging purposes

File: scripts/lib/spammer.ts (hunk 3 above)
Comment (hweawer) [reply scripts/lib/spammer.ts]: Done

File: scripts/lib/spammer.ts (hunk 2 above)
Comment (hweawer) [reply scripts/lib/spammer.ts]: Done

File: scripts/lib/spammer.ts (hunk 1 above)
Comment (hweawer) [reply scripts/lib/spammer.ts]: Done

File: monitoring/index.ts [hunk 4]
```diff
@@ -22,11 +22,11 @@ async function main() {
...
     envConfig.DATA_BUS_ADDRESS,
     [
-      "event MessageDepositV1(address indexed guardianAddress, (bytes32 depositRoot, uint256 nonce, uint256 blockNumber, bytes32 blockHash, bytes signature, uint256 stakingModuleId, (bytes32 version) app) data)",
-      "event MessagePauseV2(address indexed guardianAddress, (bytes32 depositRoot, uint256 nonce, uint256 blockNumber, bytes32 blockHash, bytes signature, uint256 stakingModuleId, (bytes32 version) app) data)",
-      "event MessagePauseV3(address indexed guardianAddress, (uint256 blockNumber, bytes signature, (bytes32 version) app) data)",
+      "event MessageDepositV1(address indexed guardianAddress, (uint256 blockNumber, bytes32 blockHash, bytes32 depositRoot, uint256 stakingModuleId, uint256 nonce, (bytes32 r, bytes32 vs) signature, (bytes32 version) app) data)",
+      "event MessagePauseV2(address indexed guardianAddress, (uint256 blockNumber, bytes32 blockHash, (bytes32 r, bytes32 vs) signature, uint256 stakingModuleId, (bytes32 version) app) data)",
+      "event MessagePauseV3(address indexed guardianAddress, (uint256 blockNumber, (bytes32 r, bytes32 vs) signature, (bytes32 version) app) data)",
```
Comment (F4ever) [inline monitoring/index.ts:27]: Add `bytes32 blockHash` here ^

**Review by F4ever — APPROVED** [review]

**Review by eddort — APPROVED** [review]

**Pull request merged**